smalltalksrc/Slang-Tests/SlangBasicTranslationTest.class.st

```diff
@@ -1321,6 +1359,96 @@ SlangBasicTranslationTest >> testSendTimes [
 	self assert: translation equals: '(1 + 2) * 3'
 ]
 
+{ #category : #'tests-send' }
+SlangBasicTranslationTest >> testSendTimesRepeat [
+
+	| translation send variable expression |
+	variable := TVariableNode new setName: 'var'.
+
+	expression := TSendNode new
+		              setSelector: #-
+		              receiver: variable
+		              arguments: { (TConstantNode value: 7) }.
+	send := TSendNode new
+		        setSelector: #timesRepeat:
+		        receiver: (TConstantNode value: 3)
+		        arguments: { (TStmtListNode new setStatements:
+				         { (TAssignmentNode new
+					          setVariable: variable
+					          expression: expression) }) }.
+	translation := self translate: send.
+
+	self assert: translation trimBoth equals: '{
+	var -= 7;
+}{
+	var -= 7;
+}{
+	var -= 7;
+}'
+]
+
+{ #category : #'tests-send' }
+SlangBasicTranslationTest >> testSendToDo [
+
+	| translation send variable expression |
+	variable := TVariableNode new setName: 'var'.
+
+	expression := TSendNode new
+		              setSelector: #-
+		              receiver: variable
+		              arguments: { (TConstantNode value: 7) }.
+	send := TSendNode new
+		        setSelector: #to:do:
+		        receiver: (TConstantNode value: 1)
+		        arguments: { 
+				        (TConstantNode value: 10).
+				        (TStmtListNode new
+					         setArguments: #( i )
+					         statements:
+					         { (TAssignmentNode new
+						          setVariable: variable
+						          expression: expression) }) }.
+	translation := self translate: send.
+
+	self assert: translation trimBoth equals: 'for(i = 1; i <= 10; i++) {
+	var -= 7;
+}'
+]
+
+{ #category : #'tests-send' }
+SlangBasicTranslationTest >> testSendToDoAvoidUnderflowing [
+
```

Comment:
What does "avoid underflowing" mean?

Reply:
ofc, that is not clear, i added a comment and changed the name. I think it is ok now :)